### ISSUE TITLE
possible bug with design matrix add_poly

### ISSUE THREAD

**Issue by maxfarrens** [cosanlab/nltools#347]
<img width="1186" alt="Screen Shot 2020-05-04 at 1 20 22 PM" src="https://user-images.githubusercontent.com/36425264/80994074-2896a580-8e0a-11ea-9d3e-63f05555b934.png">

issue appending (adds unwanted rows), using ver 0.3.19

code:
`# add motion covariates
            curr_cov = covs.iloc[time1:time2, :]
            cov = Design_Matrix(curr_cov, sampling_freq=samp_freq)
            print(cov.shape)
            
            # add polynomial regressors
            cov_poly = cov.add_poly(order=2, include_lower=True)
            print(cov_poly.shape)`

output:
`(45, 49)
(53, 52)`

**Comment by maxfarrens** [comment]
<img width="1170" alt="Screen Shot 2020-05-04 at 1 27 29 PM" src="https://user-images.githubusercontent.com/36425264/80994713-0f422900-8e0b-11ea-8996-dd880f372e51.png">

looks like the issue arises when index isn't set at 0. Is this intended behavior for this edge case or should a reset_index be added?

**Comment by ejolly** [comment]
Interesting, yes we assume that the index is numerical and starts from 0 because for most use cases it should. Other than a date-time index (which we don't formally support) it's a bit strange to slice rows and only call `Design_Matrix` methods on those rows; that was never intended usage. Was there something specific you were trying to accomplish by doing that?

I would recommend a `.reset_index` to avoid this issue, but I'm going leave this issue open and rename it in case we want to support more flexible use-cases in the future.

**Comment by ljchang** [comment]
yeah, we fixed it with `.reset_index`. I had never seen it before. If we don't do anything, it might be good to note this in the documentation that the indices should be the same length. I'm not sure how I feel about forcing a reset inside the add_poly() method (or any others), we could add a check to make sure indices match, not just length, and return a warning if not.

**Comment by ejolly** [comment]
yea I don't love the idea of forcing a `reset_index` incase the user has a meaningful index they want to retain. Definitely think we should mention it in the docs tho or add a warning if the index doesn't start at 0.

**Comment by ejolly** [comment]
Closing as obsolete. The extra-rows behavior was a pandas index-alignment artifact. In v0.6.0 `DesignMatrix` wraps a polars DataFrame, which has no row index — `add_poly` builds terms purely positionally and append handles column prefixing, not row-index alignment — so this class of misalignment can no longer occur.
